### ISSUE TITLE
Survey Editor 1.0.58 does not render Select2 and Tagbox custom widgets

### ISSUE THREAD

**Issue by willgdjones** [surveyjs/surveyjs#1456]
### Are you requesting a feature, reporting a bug or asking a question?

Reporting a bug

### What is the current behavior?

Survey Js Editor does not render [Select2](https://surveyjs.io/Examples/Library/?id=custom-widget-select2&platform=Vue&theme=default ) or [Tagbox](https://surveyjs.io/Examples/Library/?id=custom-widget-select2-tagbox&platform=jQuery&theme=default) widgets: 

When I enter the JSON from the tagbox example into the Survey JS editor (v1.0.58),
```
{
  "pages": [
    {
     "name": "page1",
      "elements": [
        {
          "type": "dropdown",
          "renderAs": "select2",
          "choicesByUrl": {
            "url": "https://restcountries.eu/rest/v2/all"
          },
          "name": "countries",
          "title": "Please select the country you have arrived from:"
        }
      ]
    }
  ]
}

```

I receive the error: 

```
Error: The property 'renderAs' in class 'dropdown' is unknown. The list of available properties are: name, visible, useDisplayValuesInTitle, visibleIf, width, startWithNewLine, indent, page, title, description, commentText, valueName, enableIf, defaultValue, correctAnswer, isRequired, requiredErrorText, readOnly, validators, titleLocation, hasComment, hasOther, choices, choicesOrder, choicesByUrl, hideIfChoicesEmpty, choicesVisibleIf, choicesEnableIf, otherText, otherErrorText, storeOthersAsComment, optionsCaption.
```

### What is the expected behavior?

It should render the Select2 and Tagbox widgets.

### How would you reproduce the current behavior (if this is a bug)?

Open the SurveyJS editor and enter the JSON

#### Provide the test code and the tested page URL (if applicable)
Hard to do unfortunately.

### Specify your

* browser: Chrome
* browser version: Version 70.0.3538.110
* surveyjs platform (angular or react or jquery or knockout or vue): Vue
* surveyjs version: 1.0.58

**Comment by andrewtelnov** [comment]
@willgdjones Yes, it stops to work with service. We will take a look.

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@willgdjones The bug has been fixed in the [custom widget repo](https://github.com/surveyjs/widgets)

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@willgdjones We will release a new version and update NPM very likely on Tuesday. I have just found another bug and have fixed it. It is related to renderAs property. Unfortunately, you would have to rebuild the surveyjs library as well.

Thank you,
Andrew

**Comment by willgdjones** [comment]
Thanks!

**Comment by andrewtelnov** [comment]
@willgdjones The new update is released. It should be fine now.

**Comment by willgdjones** [comment]
Thanks for updating - I'm actually getting a `Component survey is already registered` error in my console. This might be unrelated to the original issue though. Is there a standard way to integrate Survey with Vue? We have currently been working off this example: https://github.com/surveyjs/surveyjs_vue_quickstart

**Comment by willgdjones** [comment]
Ok - I can confirm that it seems to have been introduced in 1.0.59 somehow! It occurs when I use both SurveyVue and Survey Editor. Let me know if I can provide further details. Thanks for all your help!

**Comment by tsv2013** [comment]
@willgdjones  Can you try to remove node_modules and then reinstall them via `npm i` command?

**Comment by willgdjones** [comment]
Ok - I thought I tried that yesterday, but I just tried it now and it worked - so thank you!

**Comment by willgdjones** [comment]
I'm still getting the same errors as before, here are the package versions I have.

```
"select2": "^4.0.6-rc.1",
"survey-knockout": "1.0.59",
"survey-vue": "1.0.59",
"surveyjs-editor": "1.0.59",
"surveyjs-widgets": "1.0.59",
```

Let me know if there's anything else I need to add.

Thanks,
Will

**Comment by dmitrykurmanov** [comment]
@willgdjones Hello! I've created an example: https://plnkr.co/edit/5tNsJhUNSeuFXgBDs8yk?p=preview and it looks ok for me. Could you please edit it to reproduce the problem?

**Comment by willgdjones** [comment]
I can confirm that your versions work for me in my application when importing the packages from the various CDNs! Thank you for that. However, when I run a fresh install of my node_modules and import them in the following way:

```
import "surveyjs-editor/surveyeditor.css";
import * as SurveyEditor from "surveyjs-editor";
import * as widgets from "surveyjs-widgets";
```

Then I get the error that I had before. This is importing the files within a component of mine, and so I must be doing so incorrectly. Are there best practices for importing the necessary dependencies from inside a component itself? Thanks for your help.

**Comment by willgdjones** [comment]
https://github.com/surveyjs/surveyjs/issues/1456

**Comment by willgdjones** [comment]
The Plnkr example seems to have stopped working :/

**Comment by willgdjones** [comment]
I'm still having issues :(

**Comment by dmitrykurmanov** [comment]
@willgdjones could you please check our quickstart repo: https://github.com/surveyjs/surveyjs_vue_quickstart/blob/master/src/components/SurveyEditor.vue ? May be it will help.

**Comment by willgdjones** [comment]
Yep, that's the example we were working off. :/ We'll take a further look over the weekend but for now we'll just import the packages via the CDN.

**Comment by dmitrykurmanov** [comment]
> Yep, that's the example we were working off. :

@willgdjones oh I am sorry I've missed it.

**Comment by willgdjones** [comment]
I believe that I've managed to get this working, I needed to import Select2 along with the other packages in the Vue component:

```
import * as SurveyEditor from "surveyjs-editor";
import * as Select2 from "select2";
import * as SurveyKo from "survey-knockout";
import * as widgets from "surveyjs-widgets";
```

**Comment by dmitrykurmanov** [comment]
@willgdjones created new issue https://github.com/surveyjs/surveyjs/issues/1470

**Comment by nguyencongt3** [comment]
I get it work by downgrade to 1.0.71 version.

**Comment by andrewtelnov** [comment]
@nguyencongt3 It works fine with the latest version. I have made several example several days ago.
The latest version is v1.0.78 and we have just released v1.0.79

Thank you,
Andrew
